src/components/KPI/KPI.jsx

```diff
@@ -0,0 +1,108 @@
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+export default function KPI({ qlikApp, objectId, title, color }) {
+  // Define states
+  const [kpiObject, setKpiObject] = useState();
+
+  // Get the kpiObject, using the props provided and set to state
+  useEffect(() => {
+    // Get the kpiObject, which is an instance of a qlik object & set to state
+    const getSetObjectDetails = async () => {
+      // Get the specified object from the instance of the specific app
+      const qObject = await qlikApp.getObject(objectId);
+      setKpiObject(qObject);
+      // console.log('qObject', qObject);
+    };
+    // Only call the getSetObject when: kpiObject doesn't exist, qlikApp does exist and objectId does exist
+    if (!kpiObject && qlikApp && objectId) {
+      getSetObjectDetails();
+    }
+  }, [kpiObject, qlikApp, objectId]);
+
+  // Get the relevant kpiData from the layout of the kpiObject and set to state
+  const [kpiData, setKpiData] = useState(undefined);
+  useEffect(() => {
+    const getKpiData = async () => {
+      const qPath = '/qHyperCubeDef';
+      const qPages = [
+        {
+          qLeft: 0,
+          qTop: 0,
+          qWidth: 2, // layoutObject['qHyperCube']['qSize']['qcx']
+          qHeight: 2, // layoutObject['qHyperCube']['qSize']['qcy']
+        },
+      ];
+
+      const layout = await kpiObject.getHyperCubeData(qPath, qPages);
+      console.log('layout', layout);
+      const newKpiData = {
+        value: Math.round(layout[0].qMatrix[0][0].qNum / 10000000),
+        percent: layout[0].qMatrix[0][1].qNum * 100,
+      };
+      setKpiData(newKpiData);
+    };
+    if (kpiObject && !kpiData) {
+      getKpiData();
+    }
+  }, [kpiData, kpiObject]);
+  let arrow;
+  // Determine the direction of the arrow symbol, based on percentage value
+  if (kpiData?.percent < 0) {
+    arrow = 'down';
+  } else if (kpiData?.percent > 0) {
+    arrow = 'up';
+  } else {
+    arrow = '';
+  }
+  return (
+    <Section color={color}>
+      <h5>£ {kpiData?.value.toLocaleString()} m</h5>
```

Comment:
toLocaleString() won't automatically convert things to 'millions', so you'll have to modify it to get it to work in your situation, or maybe use another function/library (numeral is a pretty common and relatively easy to use library for formatting numbers http://numeraljs.com/#format)
I've rigged up an example, where I made some selections in the click app to get the values lower, and as you can see the React application is showing 6 million, yet in the qlik app, it's reading £63,478,000 i.e. 63million
![image](https://user-images.githubusercontent.com/84034951/163984251-7893e043-b48d-4149-a06f-16a923bb8b48.png)